### ISSUE TITLE
Cannot tap a "toggle" button in WebView

### ISSUE THREAD

**Issue by pliablepixels** [appium/python-client#340]
## The problem

I am developing an app using ionic (v1). Ionic has an `<ion-toggle>` button (that is essentially a glorified checkbox) that I am not able to toggle

## Environment
- Appium version (or git revision) that exhibits the issue: 1.11.1
- Desktop OS/version used to run Appium: OSX High Sierra 
- Node.js version (unless using Appium.app|exe): 8.11.2
- Mobile platform/version under test: Android 7.1.1
- Real device or emulator/simulator: Emulator
- Appium CLI or Appium.app|exe: appium run via NodeJS

## Details

This is the toggle I am trying to click:

![screen shot 2019-03-07 at 4 15 33 pm](https://user-images.githubusercontent.com/4116654/53992419-c6a1b980-40fa-11e9-80b2-a9b5a3d82156.png)

See code below

My template:
```
<ion-toggle id="testaut_useauth_toggle" 
            ng-change="toggleAuth()" 
            ng-model="wizard.useauth" 
            toggle-class="toggle-calm">{{'kWizUseAuth' |
          translate}}
</ion-toggle>
```

Invocation of code in appium test script:

```
self.tap_item("testaut_useauth_toggle")
```

1st implementation:

```
def click_item(self,id, wait=5):
        element = self.driver.find_element_by_id(id)
        element.click()
        sleep(wait)
````

Output:
```
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: unknown error: Element <div class="item item-toggle toggle-small ng-not-empty ng-valid" id="testaut_useauth_toggle" ng-change="toggleAuth()" ng-model="wizard.useauth" toggle-class="toggle-calm" style="">...</div> is not clickable at point (180, 190). Other element would receive the click: <section ng-show="selected" ng-class="{current: selected, done: completed}" class="step current" ng-transclude="" wz-title="2" style="">...</section>
  (Session info: chrome=55.0.2883.91)
````

2nd Implementation:

```
 def tap_item(self,id,wait=5):
        print ('Tapping item: {}'.format(id))
        element = self.driver.find_element_by_id(id)
        #self.driver.execute_script("arguments[0]).click();", element);
        self.driver.execute_script("document.getElementById(arguments[0]).click();", id);
```
Output:
No error, but nothing happens

3rd implementation:

```
def tap_item(self,id,wait=5):
        l = element.location
        s = element.size
        print ("HERE {} & {}".format(l,s))

        factor = float(3.428) # emulator VM reported size / screen size 
        tapX = 286 * factor # I loaded the image in an editor and found the center of the toggle
        tapY = 190 * factor 
        old_context = self.driver.current_context
        self.driver.switch_to.context("NATIVE_APP")
        actions = TouchAction(self.driver)
        print ("TAPPING AT {}:{}".format(str(tapX), str(tapY)))
        actions.tap(None,tapX, tapY,1)
        actions.perform()
        self.driver.switch_to.context(old_context)
        sleep(wait)
```
Output: nothing happens

**Comment by KazuCocoa** [comment]
> selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: unknown error: Element <div class="item item-toggle toggle-small ng-not-empty ng-valid" id="testaut_useauth_toggle" ng-change="toggleAuth()" ng-model="wizard.useauth" toggle-class="toggle-calm" style="">...</div> is not clickable at point (180, 190). Other element would receive the click: <section ng-show="selected" ng-class="{current: selected, done: completed}" class="step current" ng-transclude="" wz-title="2" style="">...</section>

Is this https://stackoverflow.com/questions/11908249/debugging-element-is-not-clickable-at-point-error ? (Appium throws commands to chromedriver directly)


If you could try `appium@beta`, you also can try `mobile: webatom` feature
https://github.com/appium/appium-espresso-driver/blob/36837a1a8735a7ceaa64711efecd9fd2ec5b66b6/lib/commands/general.js#L99
https://github.com/appium/appium-espresso-driver/pull/380

**Comment by pliablepixels** [comment]
Yes, the error is the same as that specific SO thread, but none of the solutions actually worked. That being said, I'm going to close this out as I figured out how to handle this element. I had to get more specific after looking at how the toggle button was implemented and after some experimentation on which sub-element needs to be the target of the click.

Final solution:

```
def tap_toggle(id,wait=Wait.TOGGLE):
    element = driver.find_element_by_id(id)
    element = element.find_element_by_tag_name('label')
    element.click()
    sleep(wait)
```
